plugin/src/main/java/org/opennms/velocloud/ApiClient.java

```diff
@@ -1,13 +1,46 @@
 package org.opennms.velocloud;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
 
-import org.opennms.velocloud.model.Alert;
+import org.opennms.velocloud.client.handler.Configuration;
+import org.opennms.velocloud.client.handler.auth.ApiKeyAuth;
+import org.opennms.velocloud.client.handler.auth.HttpBasicAuth;
+import org.opennms.velocloud.client.handler.auth.OAuth;
 import org.opennms.velocloud.model.Topology;
 
-public class ApiClient {
+public class ApiClient extends org.opennms.velocloud.client.handler.ApiClient {
+
+    public static final String AUTHORIZATION = "Authorization";
+    public static final String HEADER = "header";
+    public static final String TOKEN = "Token";
+
+    public ApiClient(final String url, final String params, final String authType) {
+        super();
+        setBasePath(url);
+        switch (authType){
+            case AuthTypes.API_KEY_AUTH:
+                authentications = Map.of("ApiKeyAuth", new ApiKeyAuth(HEADER, AUTHORIZATION));
```

Comment:
Isn't authentications prefilled already with all available auth impls?

Reply:
Only with and empty map (the below code is in the constructor for the ClientApi generated

```
    // Setup authentications (key: authentication name, value: authentication).
    authentications = new HashMap<String, Authentication>();
    // Prevent the authentications from being modified.
    authentications = Collections.unmodifiableMap(authentications);
```

Reply:
I have some values in there. They will be generated if the auth methods are provided in the swagger spec. The client needs a `mvn clean install` from time to time...

Reply:
See here: https://github.com/swagger-api/swagger-codegen-generators/blob/16520bea15357f1adc92abac0730f87c331968f7/src/main/resources/handlebars/Java/libraries/jersey2/ApiClient.mustache#L89